tests/support/kubernetes.py

```diff
@@ -1,134 +1,60 @@
-            self.objects[namespace]["ConfigMap"] = {}
-        if name in self.objects[namespace]["ConfigMap"]:
-            raise ApiException(status=500, reason=f"{namespace}/{name} exists")
-        self.objects[namespace]["ConfigMap"][name] = config_map
-
-    async def delete_namespaced_config_map(
-        self, name: str, namespace: str
-    ) -> V1Status:
-        self._maybe_error("delete_namespaced_config_map", name, namespace)
-        if namespace not in self.objects:
-            reason = f"{namespace}/{name} not found"
-            raise ApiException(status=404, reason=reason)
-        if name not in self.objects[namespace].get("ConfigMap", {}):
-            reason = f"{namespace}/{name} not found"
-            raise ApiException(status=404, reason=reason)
-        del self.objects[namespace]["ConfigMap"][name]
-        return V1Status(code=200)
-
-    # POD API
-
-    async def create_namespaced_pod(self, namespace: str, pod: V1Pod) -> None:
-        self._maybe_error("create_namespaced_pod", namespace, pod)
-        assert namespace == pod.metadata.namespace
-        name = pod.metadata.name
-        if namespace not in self.objects:
-            self.objects[namespace] = {}
-        if "Pod" not in self.objects[namespace]:
-            self.objects[namespace]["Pod"] = {}
-        if name in self.objects[namespace]["Pod"]:
-            raise ApiException(status=500, reason=f"{namespace}/{name} exists")
-        pod.status = V1PodStatus(phase="Running")
-        self.objects[namespace]["Pod"][name] = pod
-
-    async def delete_namespaced_pod(
-        self, name: str, namespace: str
-    ) -> V1Status:
-        self._maybe_error("delete_namespaced_pod", name, namespace)
-        if namespace not in self.objects:
-            reason = f"{namespace}/{name} not found"
-            raise ApiException(status=404, reason=reason)
-        if name not in self.objects[namespace].get("Pod", {}):
-            reason = f"{namespace}/{name} not found"
-            raise ApiException(status=404, reason=reason)
-        del self.objects[namespace]["Pod"][name]
-        return V1Status(code=200)
-
-    async def read_namespaced_pod(self, name: str, namespace: str) -> V1Pod:
-        self._maybe_error("read_namespaced_pod", name, namespace)
-        if namespace not in self.objects:
-            reason = f"{namespace}/{name} not found"
-            raise ApiException(status=404, reason=reason)
-        if name not in self.objects[namespace].get("Pod", {}):
-            reason = f"{namespace}/{name} not found"
-            raise ApiException(status=404, reason=reason)
-        return self.objects[namespace]["Pod"][name]
+        self.queue: Queue[str] = Queue()
+
+    @asynccontextmanager
+    async def stream(
+        self,
+        api_call: Callable[..., Awaitable[Any]],
+        *args: Any,
+        **kwargs: Any,
+    ) -> AsyncIterator[MockKubernetesStream]:
+        await self.queue.put("")
```

Comment:
Is this just putting an empty event into the stream?  Might it not be more helpful for debugging if instead you put something like the api_call's name, or below, signal_change?  Or at that point do we just not care because we're so far down in the bowels of the mock framework that the details are no longer interesting?

Reply:
We're not using the value of the element in the queue for anything. I probably should make that more explicit. It's just a signaling mechanism. We want a one-to-one correspondence between wake-ups and generation of new events, and we want it to be repeatable, so that fit a queue better than an event or a semaphore even though we don't need the value.